src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -11,25 +11,47 @@
 import com.aws.greengrass.security.exceptions.KeyLoadingException;
 import com.aws.greengrass.security.exceptions.ServiceProviderConflictException;
 import com.aws.greengrass.security.exceptions.ServiceUnavailableException;
+import com.aws.greengrass.util.EncryptionUtils;
 import lombok.AccessLevel;
 import lombok.Getter;
 
+import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.List;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
 
-public class SecurityService {
+public final class SecurityService {
     private static final Logger logger = LogManager.getLogger(SecurityService.class);
     private static final String KEY_TYPE = "keyType";
     private static final String KEY_URI = "keyUri";
+    private static final String CERT_URI = "certificateUri";
 
     @Getter(AccessLevel.PACKAGE)
     private final ConcurrentMap<CaseInsensitiveString, CryptoKeySpi> cryptoKeyProviderMap;
 
+    /**
+     * Constructor of security service.
+     */
     public SecurityService() {
         this.cryptoKeyProviderMap = new ConcurrentHashMap<>();
```

Comment:
nit: Instant initialization?

Reply:
will do. But you know compiler will put instance initialization into constructor as well.